source/jormungandr/jormungandr/equipments/sytral.py

```diff
@@ -0,0 +1,108 @@
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+from __future__ import absolute_import, print_function, unicode_literals, division
+
+from jormungandr import cache, app
+from navitiacommon import type_pb2
+from dateutil import parser
+from jormungandr.utils import date_to_timestamp
+
+import pybreaker
+import logging
+import jmespath
+
+
+class SytralProvider(object):
+    """
+    Class managing calls to SytralRT webservice, providing real-time equipment details
+    """
+
+    def __init__(self, url, timeout=2, **kwargs):
+        self.url = url + '/equipments'
+        self.timeout = timeout
+        self.breaker = pybreaker.CircuitBreaker(
+            fail_max=kwargs.get('circuit_breaker_max_fail', app.config['CIRCUIT_BREAKER_MAX_SYTRAL_FAIL']),
+            reset_timeout=kwargs.get(
+                'circuit_breaker_reset_timeout', app.config['CIRCUIT_BREAKER_SYTRAL_TIMEOUT_S']
+            ),
+        )
+
+    def get_informations(self, stop_points_list):
+        """
+        Get equipment information from Sytral webservice and update response accordingly
+        """
+        data = self._call_webservice()
+
+        if data:
+            return self._process_data(data, stop_points_list)
+
+    @cache.memoize(app.config.get(str('CACHE_CONFIGURATION'), {}).get(str('TIMEOUT_SYTRAL'), 30))
+    def _call_webservice(self):
+        """
+        Call equipments webservice with URL defined in settings
+        :return: data received from the webservice
+
+        NOTE: CALL TO ACTUAL WEBSERVICE WILL BE DONE WHEN IT WILL BE AVAILABLE (NAVP-1231)
+        """
+        logging.getLogger(__name__).debug('sytralRT RT service , call url : {}'.format(self.url))
+        pass
+
+    def _process_data(self, data, stop_points_list):
+        """
+        For each stop point in te journeys response, the structure 'equipment_details' is updated if the corresponding code is present
+        :param data: equipments data received from the webservice
+        :param stop_points_list: list of stop_points from the protobuf response
+        """
+        for st in stop_points_list:
+            for code in st.codes:
+                if "TCL_" in code.type:
```

Comment:
Just a question, it is better to have it in parameter ? I don't know if it's smarty to hard coded it